commitizen-tools/commitizen
# feat(check): Add `--allow-abort` option

## Description

Empty commit messages indicate to Git that the commit should be aborted. Displaying an error message when the commit is already being aborted typically only creates confusion. Add an `--allow-abort` argument to the check command and `allow_abort` config variable, both defaulting to false for backwards compatibility. When the commit message is empty, determine the outcome based on the `allow_abort` setting alone, ignoring the pattern.

Closes #424.

## Checklist


## Expected behavior

No error message is displayed when aborting a commit locally, unless the commit .

## Steps to Test This Pull Request

1. Run `cz commit`.
2. Save and close the commit message file that appears with an empty commit message.
3. No error occurs if you add `allow_abort = true` to your config, but an error occurs otherwise.

## Additional context

Would it make sense for `--allow-abort` to also be added as an arg to the `commit` command? How do folks feel about the argument/setting name and overall approach? See also #247 for a related feature proposal. Since the time of that discussion, I realized that naming a Commitizen argument `--allow-empty-message` could prove extremely confusing, because passing it to Commitizen would not, by itself, cause commits with empty messages to be committed like the Git argument of the same name. It would also interact poorly with the proposal to allow ferrying arbitrary arguments to `git commit` in #451, because it would be easy to accidentally pass the argument to Git, resulting in commits with empty messages no longer being aborted.
